app/models/patron.rb

```diff
@@ -218,6 +218,30 @@ def can_pay_fines?
     true
   end
 
+  def can_schedule_green_access?
+    faculty = %w[CNF MXF]
+    grad_students_and_postdocs = %w[MXD RED REG REG-SUM]
+    visiting_scholars = %w[MXAS]
+
+    [*faculty, *grad_students_and_postdocs, *visiting_scholars].include?(profile_key) ||
+      (profile_key == 'CNAC' && affiliations.include?('affiliate:fellow'))
+  end
+
+  def can_schedule_green_pickup?
+    faculty = %w[CNF MXF]
+    grad_students_and_postdocs = %w[MXD RED REG REG-SUM]
+    undergrads = %w[REU REU-SUM]
+    visiting_scholars = %w[MXAS]
+    staff = %w[CNAC CNS MXAC MXS]
+
+    [*faculty, *grad_students_and_postdocs, *undergrads, *visiting_scholars, *staff].include?(profile_key) &&
+      requests.any? { |r| %w[GREEN MEDIA-MTXT].include?(r.pickup_library) && r.ready_for_pickup? }
```

Comment:
Just thinking about this out-loud, `MEDIA-MTXT` realistically won't be the `pickup` library.  It will be the owning library, but we'll be forcing the destination to `GREEN` (and we may not want to actually show any material AT `MEDIA-MTXT` as something that you would schedule a Green paging pickup for).

If that is the case then we're in good shape WRT opening branch libraries for paging to Green (which at least one wants to do ASAP) in that we won't need to account for each one of those as they open, as long as we're forcing the pickups to `GREEN`.